File: app/src/main/java/com/tasktree/Ball.java
package com.tasktree;

import com.google.common.base.Objects;

class Ball {

    private int number;
    private Position position;

    public Ball( int number, Position position ) {
        this.number = number;
        this.position = position;
    }

    public int getNumber( ) {
        return number;
    }

    public Position getPosition( ) {
        return position;
    }

    @Override
    public boolean equals( Object obj ) {
        if ( obj == null ) return false;
        if ( getClass() != obj.getClass() ) return false;

        final Ball other = ( Ball ) obj;

        return Objects.equal( this.getNumber(), other.getNumber() )
                && Objects.equal( this.getPosition(), other.getPosition() );
    }
}

File: app/src/main/kotlin/com/tasktree/Ball.kt
package com.tasktree

import com.google.common.base.Objects

internal open class Ball(val number: Int, val position: Position) {

    override fun equals(obj: Any?): Boolean {
        if (obj == null) return false
        if (javaClass != obj.javaClass) return false

        val other = obj as Ball?

        return Objects.equal(this.number, other!!.number) && Objects.equal(this.position, other.position)
    }
}

internal class NoBall(number: Int, position: Position) : Ball(number, position)